ytypes/node.go

```diff
@@ -195,18 +194,41 @@ func retrieveNodeContainer(schema *yang.Entry, root interface{}, path *gpb.Path,
 			return retrieveNode(cschema, fv.Interface(), util.TrimGNMIPathPrefix(path, p[0:to]), np, args)
 		}
 
-		for _, p := range schPaths {
-			if !util.PathMatchesPrefix(path, p) {
-				continue
+		args := args
+		switch args.reverseShadowPath {
+		case true:
+			schPaths := util.ShadowSchemaPaths(ft)
+			for _, p := range schPaths {
+				if !util.PathMatchesPrefix(path, p) {
+					continue
+				}
+				return checkPath(p, args)
 			}
-			return checkPath(p, args)
-		}
-		for _, p := range util.ShadowSchemaPaths(ft) {
-			if !util.PathMatchesPrefix(path, p) {
-				continue
+
+			if len(schPaths) != 0 {
+				args.shadowPath = true
+			}
+			fallthrough
+		case false:
+			schPaths, err := util.SchemaPaths(ft)
+			if err != nil {
+				return nil, status.Errorf(codes.Unknown, "failed to get schema paths for %T, field %s: %s", root, ft.Name, err)
+			}
+			for _, p := range schPaths {
```

Comment:
could potentially factor out the loop into a helper function assigned immediately above the switch:

findPrefix := func (prefixes [][]string) []string {
  for _, p := range prefixes {
    if util.PathMatchesPrefix(path, p) {
      return p
    }
  }
  r

Reply:
I think this is short enough (and there are just two of them) that I didn't decide to do it. I also didn't want to reason about what happens to `args`, since it might be modified just prior to the call.